packages/db/lib/gen-migration.mjs

```diff
@@ -0,0 +1,72 @@
+import fs from 'fs'
+import { stat } from 'fs/promises'
+import path from 'path'
+import { MigrateError } from './errors.mjs'
+import loadConfig from './load-config.mjs'
+
+async function generateMigration (_args) {
+  const { configManager } = await loadConfig({
+    string: ['to'],
+    boolean: ['rollback'],
+    alias: {
+      t: 'to',
+      r: 'rollback'
+    }
+  }, _args)
+  const migrationsDir = configManager.current.migrations.dir
+
+  // Check migrations directory exists
+  await checkMigrationsDirectoryExists(migrationsDir)
+
+  try {
+    await configManager.parseAndValidate()
+    const files = await fs.promises.readdir(migrationsDir)
+
+    const nextMigrationId = getNextMigrationId(files)
+
+    const nextDoMigrationName = `${nextMigrationId.toString().padStart(3, '0')}.do.sql`
+    await generateMigrationFile(migrationsDir, nextDoMigrationName)
+
+    const nextUndoMigrationName = `${nextMigrationId.toString().padStart(3, '0')}.undo.sql`
+    await generateMigrationFile(migrationsDir, nextUndoMigrationName)
+  } catch (err) {
+    console.error(err.message)
+    process.exit(1)
+  }
+}
+
+function getNextMigrationId (migrationFileName) {
+  if (migrationFileName.length === 0) {
+    return 1
+  }
+
+  const chunks = migrationFileName.pop().split('.')
+  if (chunks.length !== 3) {
+    throw new MigrateError('Previous migration has corrupted format')
+  }
+  const migrationId = parseInt(chunks[0])
+  if (isNaN(migrationId)) {
+    throw new MigrateError('Previous migration has corrupted name')
+  }
+
+  const nextId = migrationId + 1
+  return nextId
+}
+
+async function generateMigrationFile (migrationDir, id) {
+  const newMigrationFilePath = path.join(migrationDir, id)
+  await fs.promises.appendFile(newMigrationFilePath, '')
+}
+
+// TODO How to fix the duplication ?
+async function checkMigrationsDirectoryExists (dirName) {
```

Comment:
This is a copy of the function in here:
https://github.com/stanimirovv/platformatic/blob/1e0b39b55bb16a45130a869ecbe7b02ce842f5dc/packages/db/lib/migrator.mjs

We probablly should unify them.
I tried and failed to add them to `utils.js`. But got an error that `utils.js` can't import `errors.mjs`
My initial idea here is to throw a generic error from utils, catch and rethrow the as MigrationError if needed, but need some feedback.

Reply:
go ahead and move it!